backend/src/main/java/de/neuefische/booklibrary/backend/book/BookController.java

```diff
@@ -0,0 +1,92 @@
+package de.neuefische.booklibrary.backend.book;
+
+import de.neuefische.booklibrary.backend.api.ApiBookService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.validation.Valid;
+import java.util.List;
+
+import static de.neuefische.booklibrary.backend.api.ApiBookService.DEFAULT_COVER;
+
+
+@RestController
+@RequestMapping("/api/books")
+@RequiredArgsConstructor
+public class BookController {
+
+    private final ApiBookService apiBookService;
+    private final BookService bookService;
+
+    @GetMapping
+    List<Book> getAllBooks() {
+        return bookService.getAllBooks();
+    }
+
+
+    @GetMapping("/search/{searchText}")
+    public List<Book> getAllApiBooks(@PathVariable String searchText) {
+        return apiBookService.getAllApiBooks(searchText);
+    }
+
+    @GetMapping("/isbn/{isbn}")
+    public List<Book> getApiBookByIsbn(@PathVariable @Valid String isbn) {
+        return apiBookService.getApiBookByIsbn(isbn);
+    }
+
+    @GetMapping("/rentedBooks")
+    public List<Book> getRentedBooks() {
+        String user = SecurityContextHolder
+                .getContext()
+                .getAuthentication()
+                .getName();
+        return bookService.getRentedBooks(user);
+    }
+
+
+    @PostMapping
+    @ResponseStatus(code = HttpStatus.CREATED)
+    public Book addBook(@RequestBody @Valid Book newBook) {
+        if (newBook.cover() == null) {
+            return bookService.saveBook(newBook.withCover(DEFAULT_COVER));
+        }
+        RentBookInfo rentBookInfo = new RentBookInfo("", "");
+        return bookService.saveBook(newBook.withRentBookInfo(rentBookInfo));
+    }
+
+
+    @PostMapping("/rentBook/{id}")
```

Comment:
Ich verstehe, dass `rentBook` und `returnBook` nicht ganz zu den üblichen REST-Konventionen passen (weil wir damit eine Aktion anstoßen, nicht nur Daten ablegen/laden). So ganz nach REST wird es deshalb hier nicht gehen.
Nichtsdestotrotz ein Verbesserungsvorschlag für die API:
```suggestion
    @PostMapping("/{id}/rent")
```
Damit stellst Du klar, dass hier ein Aspekt des Buchs geändert wird, das sich hinter `/api/books/{id}` verbirgt.

(gleiches gilt für die anderen POST-Endpunkte)

Reply:
Danke für Dein Verbesserungsvorschlag, das macht schon Sinn. Ich habe die URLs entsprechend angepasst.